src/Connectors/Consumer/Manager.php

```diff
@@ -57,20 +64,31 @@ public function getConsumer(): ConsumerInterface
     public function handleMessage(): void
     {
         try {
-            $response = $this->consumer->consume();
+            if (!$response = $this->consumer->consume()) {
+                $this->handleTimeOut();
+                return;
+            }
+
             $record = app(ConsumerRecord::class, compact('response'));
             $this->dispatcher->handle($record);
             $this->commit();
+        } catch (ResponseTimeoutException $exception) {
+            $this->handleTimeOut();
+            return;
         } catch (ResponseWarningException $exception) {
             $this->consumerHandler->warning($exception);
         } catch (Exception $exception) {
             $this->consumerHandler->failed($exception);
         }
+
+        $this->finished = false;
     }
 
     private function commit(): void
     {
-        if ($this->autoCommit) {
+        // when running low level consumer, we dont need
+        // to commit the messages as they've already committed.
+        if ($this->autoCommit || $this->consumer instanceof ConsumerLowLevel) {
```

Comment:
Couldn't we create a `shouldCommit(): bool` method on the `ConsumerInterface` and implement it on low level / high level consumers and just use the check here?